registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/main/java/io/mosip/registration/processor/stages/demodedupe/DemodedupeProcessor.java

```diff
@@ -1,609 +1,726 @@
+			}
+		    else {
+				regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+						registrationId, "Duplicate request received for same latest transaction id. This will be ignored.");
+				object.setIsValid(false);
+				object.setInternalError(true);
+			}
+		}
+
+		return object;
+	}
+
+
+	/**
+	 * Perform demo dedupe.
+	 *
+	 * @param registrationStatusDto the registration status dto
+	 * @param object                the object
+	 * @param description
+	 * @return true, if successful
+	 * @throws IOException
+	 * @throws DataShareException
+	 * @throws JsonProcessingException
+	 * @throws PacketManagerException
+	 * @throws ApisResourceAccessException
+	 */
+	private List<DemographicInfoDto> performDemoDedupe(InternalRegistrationStatusDto registrationStatusDto,
+			MessageDTO object, LogDescription description, TrimExceptionMessage trimExceptionMessage)
+			throws ApisResourceAccessException, PacketManagerException,
+			JsonProcessingException, DataShareException, IOException {
+		String registrationId = registrationStatusDto.getRegistrationId();
+		// Potential Duplicate Ids after performing demo dedupe
+		List<DemographicInfoDto> duplicateDtos = demoDedupe.performDedupe(registrationStatusDto.getRegistrationId());
+		List<String> uniqueRIDs = new ArrayList<>();
+		List<String> matchedRegIds;
+		if (!duplicateDtos.isEmpty()) {
+			duplicateDtos.removeIf(duplicateDto -> duplicateDto.getRegId().equals(registrationId));
+			matchedRegIds = duplicateDtos.stream().map(DemographicInfoDto::getRegId)
+					.collect(Collectors.toList());
+			uniqueRIDs = abisHandlerUtil.getUniqueProcessedRecords(registrationStatusDto.getRegistrationId(),
+					SyncTypeDto.NEW.toString(), ProviderStageName.DEMO_DEDUPE, matchedRegIds);
+		}
+		if (!duplicateDtos.isEmpty() && !uniqueRIDs.isEmpty()) {
+			try {
+				biometricValidation(registrationStatusDto);
+				isMatchFound = true;
+			registrationStatusDto
+					.setLatestTransactionStatusCode(RegistrationTransactionStatusCode.IN_PROGRESS.toString());
+			registrationStatusDto.setStatusCode(RegistrationStatusCode.PROCESSING.toString());
+			registrationStatusDto.setStatusComment(StatusUtil.POTENTIAL_MATCH_FOUND.getMessage());
+			registrationStatusDto.setSubStatusCode(StatusUtil.POTENTIAL_MATCH_FOUND.getCode());
+			object.setMessageBusAddress(MessageBusAddress.ABIS_HANDLER_BUS_IN);
+			description.setCode(PlatformSuccessMessages.RPR_PKR_DEMO_DE_DUP.getCode());
+			description.setMessage(DemoDedupeConstants.RECORD_INSERTED_FROM_ABIS_HANDLER + " -- " + registrationId);
+			regProcLogger.info(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					registrationStatusDto.getRegistrationId(), DemoDedupeConstants.RECORD_INSERTED_FROM_ABIS_HANDLER);
+		} catch (BiometricRecordValidationException e) {
+			if (demodedupeMatchDecesion.equalsIgnoreCase(DemoDedupeConstants.MARK_AS_DEMODEDUPE_SUCCESS)) {

```

Comment:
should we keep this configuration separate for infant and others to provide flexibility for countries to handle infant without biometrics vs other with exceptions ?

Reply:
I have already added test cases with comment saying //Child packet Testing for same scenario

Reply:
Added seperate test cases

Reply:
fixed